pkg/dbs/studydb/db_study-surveys.go

```diff
@@ -43,12 +43,20 @@ func (dbService *StudyDBService) CreateSurveyDefintionIndexForStudy(instanceID s
 	ctx, cancel := dbService.getContext()
 	defer cancel()
 
-	_, err := dbService.collectionRefStudySurveys(instanceID, studyKey).Indexes().CreateOne(
-		ctx, mongo.IndexModel{
-			Keys: bson.D{
-				{Key: "surveyDefinition.key", Value: 1},
-				{Key: "unpublished", Value: 1},
-				{Key: "published", Value: -1},
+	_, err := dbService.collectionRefStudySurveys(instanceID, studyKey).Indexes().CreateMany(
+		ctx, []mongo.IndexModel{
+			{
+				Keys: bson.D{
+					{Key: "surveyDefinition.key", Value: 1},
+					{Key: "unpublished", Value: 1},
+					{Key: "published", Value: -1},
+				},
+			},
+			{
+				Keys: bson.D{
+					{Key: "surveyDefinition.key", Value: 1},
+					{Key: "published", Value: 1},
```

Comment:
if this index is meant to cover the possible queries in https://github.com/influenzanet/study-service/pull/13#discussion_r1003186366 (when also the sorting on published will be added) we should define them in different order:

```
{Key: "published", Value: 1},
{Key: "surveyDefinition.key", Value: 1}
```

Reply:
Thanks. Order is changed now according to this.